src/coreclr/jit/emitxarch.cpp

```diff
@@ -8872,6 +8872,42 @@ void emitter::emitDispIns(
                 { // (val < 0)
                     printf("-0x%IX", -val);
                 }
+
+                unsigned iconType = id->idDebugOnlyInfo()->idFlags & GTF_ICON_HDL_MASK;
+                if ((val != 0) && id->idIsLargeCns())
+                {
+                    if (iconType == GTF_ICON_CLASS_HDL)
+                    {
+                        printf(" ; %.32s", emitComp->eeGetClassName(reinterpret_cast<CORINFO_CLASS_HANDLE>(val)));
+                    }
+                    else if (iconType == GTF_ICON_STR_HDL)
+                    {
+                        const WCHAR* str = emitComp->eeGetCPString(static_cast<size_t>(val));
+                        if (str != nullptr)
+                        {
+                            size_t    len            = wcslen(str);
+                            const int maxLength      = 32;
+                            WCHAR     buf[maxLength] = {0};
+                            wcsncpy(buf, str, min(maxLength, len));
+                            for (size_t i = 0; i < min(maxLength, len); i++)
+                            {
+                                // Escape \n and \r symbols
+                                if (buf[i] == L'\n' || buf[i] == L'\r')
+                                {
+                                    buf[i] = ' ';
+                                }
+                            }
+                            if (len > maxLength)
```

Comment:
```suggestion
                            if (len > maxLength - 1)
```
and you need to modify at `-4`, `-3`, and `-2`, leaving `-1` as null.